applications/nrf5340_audio/src/audio/audio_datapath.c

```diff
@@ -912,34 +915,57 @@ void audio_datapath_stream_out(struct net_buf *audio_frame)
 		return;
 	}
 
+	/* Upon first received audio frame, the delta will be invalid (as there is no previous value
+	 * to compare it to). Hence, this function only prints LOG_ERR if there are consecutive
+	 * errors.
+	 */
+	static uint32_t consec_invalid_ts_deltas;
 	bool sdu_ref_not_consecutive = false;
 
-	if (ctrl_blk.prev_pres_sdu_ref_us) {
-		uint32_t sdu_ref_delta_us = meta->ref_ts_us - ctrl_blk.prev_pres_sdu_ref_us;
-
-		/* Check if the delta is from two consecutive frames */
-		if (sdu_ref_delta_us <
-		    (CONFIG_AUDIO_FRAME_DURATION_US + (CONFIG_AUDIO_FRAME_DURATION_US / 2))) {
-			/* Check for invalid delta */
-			if ((sdu_ref_delta_us >
-			     (CONFIG_AUDIO_FRAME_DURATION_US + SDU_REF_CH_DELTA_MAX_US)) ||
-			    (sdu_ref_delta_us <
-			     (CONFIG_AUDIO_FRAME_DURATION_US - SDU_REF_CH_DELTA_MAX_US))) {
-				LOG_DBG("Invalid sdu_ref_us delta (%d) - Estimating sdu_ref_us",
-					sdu_ref_delta_us);
-
-				/* Estimate sdu_ref_us */
-				meta->ref_ts_us = ctrl_blk.prev_pres_sdu_ref_us +
-						  CONFIG_AUDIO_FRAME_DURATION_US;
-			}
+	uint32_t sdu_ref_delta_us = meta->ref_ts_us - ctrl_blk.prev_pres_sdu_ref_us;
+
+	/* Check if the new timestamp is consective wrt. the previous timestamp */
+	if (sdu_ref_delta_us > CONSECUTIVE_TS_LIMIT_US) {
+		if (consec_invalid_ts_deltas) {
+			LOG_ERR("sdu_ref_us not from consecutive frames (delta: %d us)",
+				sdu_ref_delta_us);
 		} else {
-			LOG_INF("sdu_ref_us not from consecutive frames (diff: %d us)",
+			LOG_DBG("sdu_ref_us not from consecutive frames (delta: %d us)",
 				sdu_ref_delta_us);
-			sdu_ref_not_consecutive = true;
 		}
-	}
 
-	ctrl_blk.prev_pres_sdu_ref_us = meta->ref_ts_us;
+		sdu_ref_not_consecutive = true;
+		ctrl_blk.prev_pres_sdu_ref_us = meta->ref_ts_us;
+		consec_invalid_ts_deltas++;
+		/* Check if the new timestamp is within the expected bounds */
+	} else if (IN_RANGE(sdu_ref_delta_us,
+			    CONFIG_AUDIO_FRAME_DURATION_US + SDU_REF_CH_DELTA_MAX_US,
+			    CONFIG_AUDIO_FRAME_DURATION_US - SDU_REF_CH_DELTA_MAX_US)) {
+		if (consec_invalid_ts_deltas) {
```

Comment:
Would this mean, "not from consecutive" should only pop up one time?

Reply:
Correct